src/mds/MDSContext.h

```diff
@@ -135,11 +137,36 @@ class C_IO_Wrapper : public MDSIOContext
   C_IO_Wrapper(MDSRank *mds_, MDSInternalContextBase *wrapped_) : MDSIOContext(mds_), wrapped(wrapped_) {
     assert(wrapped != NULL);
   }
+
+  ~C_IO_Wrapper() {
+    delete wrapped;
+  }
+
   virtual void finish(int r) {
     wrapped->complete(r);
```

Comment:
Since we're now calling delete in the constructor, I think you need to do "wrapped = nullptr;" after calling complete